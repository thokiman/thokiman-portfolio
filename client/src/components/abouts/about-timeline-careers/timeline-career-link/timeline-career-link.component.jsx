import React from "react";

import TimelineCareerLinkBelow from "../timeline-career-link-below/timeline-career-link-below.component";
import TimelineCareerLinkUpper from "../timeline-career-link-upper/timeline-career-link-upper.component";

export const TimelineCareerLink = (props) => {
  if (props.placement === "upper") {
    return <TimelineCareerLinkUpper {...props} />;
  }
  return <TimelineCareerLinkBelow {...props} />;
};

export default TimelineCareerLink;
