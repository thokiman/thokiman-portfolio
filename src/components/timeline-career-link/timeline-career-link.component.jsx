import React from "react";
import { Link } from "react-scroll";

import "./timeline-career-link.styles.scss";

const TimelineCareerLink = ({
  TimelineIcon,
  year,
  placement,
  isClick,
  yearClick,
}) => {
  return (
    <div>
      {placement === "upper" ? (
        <div className={`year-link-${placement}`}>
          <Link
            activeClass="active"
            to={year}
            spy={true}
            smooth={true}
            duration={800}
            className={
              isClick ? `year-box-${placement}-active` : `year-box-${placement}`
            }
            onClick={() => yearClick()}
          >
            {year}
          </Link>
          <TimelineIcon
            className={
              isClick
                ? `timeline-pointer-active-${placement}`
                : `timeline-pointer-${placement}`
            }
          />
        </div>
      ) : (
        <div className={`year-link-${placement}`}>
          <TimelineIcon
            className={
              isClick
                ? `timeline-pointer-active-${placement}`
                : `timeline-pointer-${placement}`
            }
          />
          <Link
            activeClass="active"
            to={year}
            spy={true}
            smooth={true}
            duration={800}
            className={
              isClick ? `year-box-${placement}-active` : `year-box-${placement}`
            }
            onClick={() => yearClick()}
          >
            {year}
          </Link>
        </div>
      )}
    </div>
  );
};

export default TimelineCareerLink;
