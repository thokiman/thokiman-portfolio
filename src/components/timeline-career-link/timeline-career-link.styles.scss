.year-link-upper {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    .year-box-upper {

        width: fit-content;
        font-size: 200%;
        border-radius: 0px 20px 0px 20px;
        background-color: rgba($color: #9096ae, $alpha: 0.4);
        padding: 5px;
        color: #e3e1e4;

        &:hover {
            animation: pulsate-bck 0.5s ease-in-out infinite both;
        }
    }

    .timeline-pointer-upper {
        position: relative;
        top: 20px;

        color: #9096ae;
        height: 40px;
        width: 40px;
        transition: top 0.2s ease-in-out, color 0.5s ease-in-out, height, width 0.5s ease-in-out;
        animation: color-change-2x 2s linear infinite alternate both;
        cursor: default;
    }

    .year-box-upper-active {
        position: relative;
        top: 45px;
        right: 20px;

        font-size: 200%;
        border: none;
        background-color: none;
        color: #3f4249;
        transition: color, border, background-color 0.5s linear;
        animation: rotate-90-bl-ccw 0.4s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;

        &:hover {
            color: #e3e1e4;

        }

    }

    .timeline-pointer-active-upper {
        position: relative;
        top: -35px;

        color: #3f4249;
        height: 90px;
        width: 90px;
        transition: top 0.2s ease-in-out, color 0.5s ease-in-out, height, width 0.5s ease-in-out;
        cursor: default;

    }
}



.year-link-below {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    position: relative;
    left: 180px;

    .year-box-below {
        position: relative;
        top: 20px;
        width: fit-content;
        font-size: 200%;
        border-radius: 0px 20px 0px 20px;
        background-color: rgba($color: #9096ae, $alpha: 0.4);
        padding: 5px;
        color: #e3e1e4;

        &:hover {
            animation: pulsate-bck 0.5s ease-in-out infinite both;
        }

    }

    .timeline-pointer-below {


        color: #9096ae;
        height: 40px;
        width: 40px;
        transition: top 0.2s ease-in-out, color 0.5s ease-in-out, height, width 0.5s ease-in-out;
        animation: color-change-2x 2s linear infinite alternate both;
        cursor: default;
    }

    .year-box-below-active {
        position: relative;
        top: -35px;
        font-size: 200%;
        border: none;
        background-color: none;
        color: #3f4249;
        transition: color, border, background-color 0.5s linear;
        animation: rotate-90-tr-ccw 0.4s cubic-bezier(0.250, 0.460, 0.450, 0.940) both;
        position: relative;

        cursor: pointer;

        &:hover {
            color: #e3e1e4;
        }

    }

    .timeline-pointer-active-below {
        position: relative;
        top: 45px;
        right: 20px;

        color: #3f4249;
        height: 90px;
        width: 90px;
        transition: top 0.2s ease-in-out, color 0.5s ease-in-out, height, width 0.5s ease-in-out;
        cursor: default;

    }
}