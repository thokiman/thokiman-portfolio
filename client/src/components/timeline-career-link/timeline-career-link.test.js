import React from "react";
import { shallow } from "enzyme";

import { TimelineCareerLink } from "./timeline-career-link.component";

describe("TimelineCareerLink component", () => {
  let wrapper;
  const mockText = "upper";

  beforeEach(() => {
    const mockProps = {
      placement: mockText,
    };
    wrapper = shallow(<TimelineCareerLink {...mockProps} />);
  });

  it("should render TimelineCareerLink component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
