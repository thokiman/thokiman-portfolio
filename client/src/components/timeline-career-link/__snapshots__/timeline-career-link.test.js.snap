// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineCareerLink component should render TimelineCareerLink component 1`] = `ShallowWrapper {}`;
